### PR TITLE
Removed deprecated Workflow methods and types from DaprClient and tests

### PR DESCRIPTION
**Pull request by WhitWaldo**
# Description

Removed the Workflow methods from the DaprClient, DaprGrpcClient and an end-to-end test as well as a few now-unused types. Also refactored the `WorkflowRuntimeStatus` enum out from `Dapr.Client` and into `Dapr.Workflow` as it has no more purpose in the former project.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1344 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Extended the documentation

### CODE REVIEW

**Comment by m3nax** [comment]
@WhitWaldo This is pr is related with the incremental source generator for actors https://github.com/dapr/dotnet-sdk/pull/1334?

**Comment by WhitWaldo** [comment]
@m3nax Nope, apologies - meant to link 1344 and accidentally put 1334 (and did it again while typing this message).

File: src/Dapr.Workflow/WorkflowRuntimeStatus.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@
 // limitations under the License.
 // ------------------------------------------------------------------------
 
-namespace Dapr.Client
+namespace Dapr.Workflow
```
Comment (philliphoff) [inline src/Dapr.Workflow/WorkflowRuntimeStatus.cs:14]: Will need to mark this as a breaking change in the release notes.

**Review by philliphoff — APPROVED** [review]

**Pull request closed without merging**